File: practicum-java/src/main/java/ru/practicum/kafka/streams/model/ChatMessage.java
package ru.practicum.kafka.streams.model;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Сообщение чата — единый формат для топиков messages и filtered_messages.
 *
 * <p>Имена полей заданы заданием 2 (ksqlDB), и задание 1 использует ту же схему:
 * один формат на обе части работы избавляет от конвертации между ними.
 * Отправитель называется {@code user_id}, а не {@code sender_id}, именно поэтому.
 *
 * <p>{@code timestamp} — строка ISO-8601, а не число: так сообщение читается глазами
 * в логах и в Kafka UI. Оконных агрегаций в работе нет, поэтому разбирать её
 * в метку времени Kafka незачем.
 *
 * <p>Record, а не класс: значение неизменяемо, а Jackson собирает его через
 * канонический конструктор.
 */
public record ChatMessage(
        @JsonProperty("user_id") String userId,
        @JsonProperty("recipient_id") String recipientId,
        @JsonProperty("message") String message,
        @JsonProperty("timestamp") String timestamp) {

    /** Копия сообщения с заменённым текстом — результат работы цензора. */
    public ChatMessage withMessage(String newMessage) {
        return new ChatMessage(userId, recipientId, newMessage, timestamp);
    }
}

File: practicum-dotnet/src/Practicum.Kafka.Streams/Models/ChatMessage.cs
using System.Text.Json.Serialization;

namespace Practicum.Kafka.Streams.Models;

/// <summary>
/// Сообщение чата — единый формат для топиков messages и filtered_messages.
///
/// Имена полей заданы заданием 2 (ksqlDB), и задание 1 использует ту же схему:
/// один формат на обе части работы избавляет от конвертации между ними.
/// Отправитель называется user_id, а не sender_id, именно поэтому.
///
/// Timestamp — строка ISO-8601, а не число: так сообщение читается глазами
/// в логах и в Kafka UI. Оконных агрегаций в работе нет, поэтому разбирать её
/// в метку времени Kafka незачем.
/// </summary>
public sealed record ChatMessage
{
    [JsonPropertyName("user_id")]
    public string UserId { get; init; } = string.Empty;

    [JsonPropertyName("recipient_id")]
    public string RecipientId { get; init; } = string.Empty;

    [JsonPropertyName("message")]
    public string Message { get; init; } = string.Empty;

    [JsonPropertyName("timestamp")]
    public string Timestamp { get; init; } = string.Empty;
}
